packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -132,6 +140,16 @@ function GridFilterForm(props: GridFilterFormProps) {
 
   const currentOperator = getCurrentOperator();
 
+  useEnhancedEffect(() => {
+    if (toFocus && toFocus.value) {
+      if (valueInputRef.current) {
+        valueInputRef.current!.focus();
+      } else {
+        filterSelectorRef.current!.focus();
+      }
```

Comment:
If the user provided a custom component (e.g. [Slider](https://mui.com/components/slider/)), which doesn't support `inputRef` as well as the control to focus is a inner element, calling `filterSelectorRef.current!.focus()` won't work.

If think we could add an imperative method `focus()` on each filter component so the task of focusing the right element is delegated to them. As an example: https://github.com/mui-org/material-ui/blob/cd247a197b772b74d81db70e1c3dc5d0a019c447/packages/mui-material/src/Select/SelectInput.js#L156-L166

Reply:
`filterSelectorRef` is a ref on the selector of the filter type which is a `Select` component provided by ourselves. I did not think about custom components, but about filters without input like "is empty"

![image](https://user-images.githubusercontent.com/45398769/141457959-b0ed326c-16d3-45cf-ac45-983e0c26dd9f.png)

Reply:
With custom filter operators that don't hold `inputRef` it doesn't work correctly:

https://deploy-preview-3004--material-ui-x.netlify.app/components/data-grid/filtering/#create-a-custom-operator

_Ignore the jump when opening the panel, it's a different bug._

<img width="400" src="https://user-images.githubusercontent.com/42154031/141486846-3789ffb4-dbf3-45c3-bf46-2d2f3ee1397e.gif" alt="s8J3hWrClL" style="max-width: 100%;">

I was proposing to add the imperative method on the filter's `InputComponent` so users can set which element to focus. 

For the situation where a filter may not require a value, you can check if it has a `InputComponent` defined and then move the focus to the operator selector instead.

https://github.com/mui-org/material-ui-x/blob/a2b1f5dc4b031cd041e9ce55953308076362221a/packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts#L104